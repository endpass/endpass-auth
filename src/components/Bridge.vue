<template>
  <v-frame>
    <p>Endpass Auth bridge {{ version }}</p>
  </v-frame>
</template>

<script>
import pkg from '@/../package.json';
import VFrame from './VFrame.vue';

if (!ENV.notShowVersion) {
  console.info(
    `%cEndpass Auth Bridge (${pkg.version}) loaded ðŸ”Œ`,
    'font-size: 14px; font-weight: bold',
  );
}

export default {
  name: 'Bridge',

  computed: {
    version() {
      return `v${pkg.version}`;
    },
  },

  components: {
    VFrame,
  },
};
</script>
