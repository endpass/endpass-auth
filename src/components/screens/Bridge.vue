<template>
  <v-frame @close="handleClose">
    <p>Endpass Auth bridge v{{ version }}</p>
  </v-frame>
</template>

<script>
import pkg from '@/../package.json';
import VFrame from '@/components/common/VFrame';
import { mapActions } from 'vuex';

if (!ENV.notShowVersion) {
  console.info(
    `%cEndpass Auth Bridge (${pkg.version}) loaded ðŸ”Œ`,
    'font-size: 14px; font-weight: bold',
  );
}

export default {
  name: 'Bridge',
  data() {
    return {
      version: pkg.version,
    };
  },

  methods: {
    ...mapActions(['dialogClose']),

    handleClose() {
      this.dialogClose();
    },
  },

  components: {
    VFrame,
  },
};
</script>
