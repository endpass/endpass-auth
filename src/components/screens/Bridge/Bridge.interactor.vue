<template>
  <div>
    <slot
      :dialog-close="dialogClose"
      :version="version"
    />
  </div>
</template>

<script>
import { useBridge } from './useBridge.composable';

export default {
  name: 'BridgeInteractor',

  setup() {
    const { dialogClose, version } = useBridge();

    return {
      dialogClose,
      version,
    };
  },
};
</script>
