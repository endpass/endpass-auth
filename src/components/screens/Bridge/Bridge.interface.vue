<template>
  <bridge-interactor #default="{ version, dialogClose }">
    <bridge-view
      :version="version"
      @close="dialogClose"
    />
  </bridge-interactor>
</template>

<script>
import BridgeInteractor from './Bridge.interactor';
import BridgeView from './Bridge.view';

export default {
  name: 'BridgeInterface',

  components: {
    BridgeView,
    BridgeInteractor,
  },
};
</script>
