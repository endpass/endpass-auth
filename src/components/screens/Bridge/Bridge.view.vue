<template>
  <v-frame @close="onClose">
    <v-description>{{ $t('global.version') }}{{ version }}</v-description>
    <div
      class="bridge-loading-container"
      data-test="bridge-loader"
    >
      <v-progress-circle
        :progress="25"
        :line-thickness="1"
      />
    </div>
  </v-frame>
</template>

<script>
import VProgressCircle from '@endpass/ui/kit/VProgressCircle';
import VFrame from '@/components/common/VFrame';
import VDescription from '@/components/common/VDescription';

export default {
  name: 'BridgeView',

  props: {
    version: {
      type: String,
      required: true,
    },
  },

  setup(props, context) {
    const onClose = () => context.emit('close');

    return {
      onClose,
    };
  },

  components: {
    VFrame,
    VDescription,
    VProgressCircle,
  },
};
</script>

<style scoped lang="postcss">
.bridge-loading-container {
  display: flex;
  align-items: center;
  flex-direction: column;
  margin: 20px;
}
</style>
