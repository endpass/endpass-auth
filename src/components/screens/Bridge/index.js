export { default } from './Bridge.interface';
