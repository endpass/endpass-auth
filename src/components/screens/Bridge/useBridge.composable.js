import { coreStore } from '@/store';
import pkg from '../../../../package.json';

if (ENV.VUE_APP_SHOW_VERSION_INFO) {
  // eslint-disable-next-line no-console
  console.info(
    `%cEndpass Auth Bridge (${pkg.version}) loaded ðŸ”Œ`,
    'font-size: 14px; font-weight: bold',
  );
}

export const useBridge = () => {
  const dialogClose = () => coreStore.dialogClose();
  const { version } = pkg;

  return {
    dialogClose,
    version,
  };
};
