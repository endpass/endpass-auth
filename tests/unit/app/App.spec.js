import Vuex from 'vuex';
import VueRouter from 'vue-router';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import Bridge from '@/App';

const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(VueRouter);

describe('App-root', () => {
  let wrapper;
  let store;
  let coreModule;

  beforeEach(() => {
    coreModule = {
      actions: {
        init: jest.fn(),
      },
      getters: {
        isDialog: jest.fn(),
      },
    };
    const router = new VueRouter();
    store = new Vuex.Store({
      router,
      modules: {
        core: coreModule,
      },
    });

    wrapper = shallowMount(Bridge, {
      store,
      localVue,
    });
  });

  describe('App render', () => {
    it('should render "empty" markup', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('App behavior', () => {
    it('should subscribe on bride after create', () => {
      expect(coreModule.actions.init).toBeCalled();
    });
  });
});
