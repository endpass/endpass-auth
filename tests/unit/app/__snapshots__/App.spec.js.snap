// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App-root App render should render "empty" markup 1`] = `
<theme-provider-stub theme="default">
  <loading-screen-stub>
    <layout-screen-stub></layout-screen-stub>
  </loading-screen-stub>
</theme-provider-stub>
`;
